compiler/src/main/scala/dx/compiler/WorkflowCompiler.scala

```diff
@@ -167,6 +176,154 @@ case class WorkflowCompiler(extras: Option[Extras],
+  private def reportableDependenciesFromApplication(
+      applet: DxApplet
+  ): Option[String] = {
+    val description = applet.describe(Set(Field.Name, Field.Details, Field.Hidden))
+
+    // Do not generate description section if hidden
+    if (description.hidden.exists(h => h)) {
+      return None
+    }
+
+    // Generate Markdown description section
+    val headerMd = s"\n## Task ${description.name}"
+    val md = description.details match {
+      case Some(JsObject(d)) => {
+        val filesMd = d.get(Constants.FileDependencies) match {
+          case Some(JsArray(a)) =>
+            s"${listMd("Files")}${a.map(v => listMd2(v.toString)).mkString}"
+          case _ => ""
+        }
+        val networkDockerImageMd = d.get(Constants.NetworkDockerImage) match {
+          case Some(JsString(s)) => s"${listMd("Network Docker image")}${listMd2(s)}"
+          case _                 => ""
+        }
+        val dynamicDockerImageMd = d.get(Constants.DynamicDockerImage) match {
+          case Some(JsBoolean(b)) if b => listMd("Docker image determined at runtime")
+          case _                       => ""
+        }
+        val staticInstanceTypeMd = d.get(Constants.StaticInstanceType) match {
+          case Some(JsString(s)) => s"${listMd("Hard-coded instance type")}${listMd2(s)}"
+          case _                 => ""
+        }
+        s"${filesMd}${networkDockerImageMd}${dynamicDockerImageMd}${staticInstanceTypeMd}"
+      }
+      case _ => ""
+    }
+    if (md.nonEmpty) {
+      Some(s"${headerMd}${md}")
+    } else {
+      None
+    }
+  }
+
+  // TODO: Use templates for Markdown dependency report
+
+  // Summarize dependencies that are not bundled for cloning
+  // with the workflow for adding to description in Markdown
+  private def summarizeReportableDependencies(
+      workflow: Workflow,
+      executableDict: Map[String, CompiledExecutable],
+      execTree: JsObject,
+      nativeAppDependencies: Vector[String],
+      fileDependencies: Vector[String],
+      dockerRegistryCredentialsUri: Option[String]
+  ): Option[String] = {
+    val headerMd =
+      "# This workflow requires access to the following dependencies that are not " +
+        "packaged with the workflow"
+
+    val topLevelMd = Vector(
+        Option.when(nativeAppDependencies.nonEmpty)(
+            Vector(listMd("Native apps")) ++ nativeAppDependencies.map(v => listMd2(v.toString))
+        ),
+        Option.when(fileDependencies.nonEmpty)(
+            Vector(listMd("Files")) ++ fileDependencies.map(v => listMd2(v.toString))
+        ),
+        Option.when(dockerRegistryCredentialsUri.nonEmpty)(
+            Vector(listMd("Docker registry credentials file"),
+                   listMd2(dockerRegistryCredentialsUri.get))
+        )
+    ).flatten.flatten.mkString
+
+    val executableNames = ExecutableTree.extractExecutableNames(execTree)
+    val executables = executableNames.collect { n =>
```

Comment:
when the function you're mapping over takes a single argument, you can shorten it to `executableNames.collect(executableDict.get)`

Reply:
```executableNames.collect(executableDict.get(_)).flatten```